File: Dec2020/14Dec/ArjitAndPrintingPress.java
/*
Problem Name: Arjit and Printing Press
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/arjit-and-printing-press/
*/
import java.util.*;
import java.io.*;
class TestClass {
    public static void main(String args[] ) throws Exception {
        Scanner in = new Scanner(System.in);
        PrintWriter out = new PrintWriter(System.out);
        int t = in.nextInt();
        for(;t>0;t--){
            char[] w = in.next().toCharArray(), r = in.next().toCharArray();
            Arrays.sort(r);
            int li = 0, ri = 0, wlen = w.length, rlen = r.length;
            for(;li<wlen && ri<rlen;){
                if(w[li] > r[ri]){
                    w[li] = r[ri];
                    li++;
                    ri++;
                }
                else
                    li++;
            }
            out.println(new String(w));
        }
        out.flush();
    }
}

File: Dec2020/14Dec/ArjitAndPrintingPress.cpp
/*
Problem Name: Arjit and Printing Press
Problem Link: https://www.hackerearth.com/practice/algorithms/greedy/basics-of-greedy-algorithms/practice-problems/algorithm/arjit-and-printing-press/
*/

#include<bits/stdc++.h>
using namespace std;

#define test(t)     int t; cin>>t; while(t--)

int main(){
    ios::sync_with_stdio(0);    
    cin.tie(0);             
    test(t){
        string word, reserve;
        cin>>word>>reserve;
        
        sort(reserve.begin(), reserve.end());
        int i = 0, j = 0;
        while(i < word.size() && j < reserve.size()){
            if(word[i] > reserve[j]){
                word[i] = reserve[j];
                j++;
            }
            i++;
        }
        cout<<word<<"\n";
    }
    return 0;
}
